### DISCUSSION TITLE
Consistency between generated molecular structure png file and atom positions

### DISCUSSION THREAD

**Discussion by HesamGit** [rdkit/rdkit#4150] [Q&A] [answered]
Is there a one-to-one and known relation between (Chem.rdDepictor.Compute2DCoords(mol)) and the generated 2d png file (Chem.Draw.MolToImageFile). For example, can we figure out the pixel in image corresponding to an atom?
It is very hard to do it manually by some coefficient as shown below and perfect math doesn't occur.

**Comment by greglandrum** [comment, marked as answer]
Hi @HesamGit, 
You can do this by using the `MolDraw2DCairo` object which `MolToImageFile` uses. Here's sample code:
```
m = Chem.MolFromSmiles('CCCOC')
rdDepictor.Compute2DCoords(m)
d2d = Draw.MolDraw2DCairo(350,300)
d2d.DrawMolecule(m)
d2d.FinishDrawing()
d2d.WriteDrawingText('foo.png')

conf = m.GetConformer()

# we can get the draw location of atom coordinates directly with atom indices:
for i in range(m.GetNumAtoms()):
    p = conf.GetAtomPosition(i)
    drawp = d2d.GetDrawCoords(i)
    print('Mol coords:','%.3f,%.3f'%(p.x,p.y),"Draw coords: ",'%.3f,%.3f'%(drawp.x,drawp.y) )
```
and a gist with a demo:
https://gist.github.com/greglandrum/0c8b80825826392a6519af7519862baf

**Reply by HesamGit** [reply]
Many thanks @greglandrum ! This feature is nice. I wonder if the opposite is possible or not. For example, if the inputs are 2d coordinates (e.g. atom positions, types and connections), would it be possible to generate Mol?

**Reply by greglandrum** [reply]
Sure. That is, of course, a totally different problem, but you can certainly start with an empty molecule and add atoms and bonds to that. Here's one discussion that deals with that: https://github.com/rdkit/rdkit/discussions/4036

**Reply by HesamGit** [reply]
Perfect! Thanks a lot.
